backend/services/implementations/testSessionService.ts

```diff
@@ -57,6 +57,27 @@ class TestSessionService implements ITestSessionService {
     return testSessionDtos;
   }
 
+  async getTestSessionsByTeacherId(
+    teacherId: string,
+  ): Promise<Array<TestSessionResponseDTO>> {
+    let testSessionDtos: Array<TestSessionResponseDTO> = [];
+    let field = "teacher";
```

Comment:
I was wondering if it made sense to combine `getTestSessionsByTeacherId`, `getTestSessionsBySchoolId` and `getTestSessionsByTestId` and add a second param for the `field` which is either "teacher", "school" or "test" -- since the code for all three is the same. or is it better to keep them separate?

Reply:
also they're all done already so idk if refactoring is too extra lol

Reply:
hmmm. i think either, or is probably fine. the way it is now probably incurs a bit of code repetition, but makes the code a bit easier to skim at first glance i think.

Reply:
thanks for thinking about this though!! i love that you're always thinking about ways to improve our codebase (: